subprojects/core/src/main/java/org/gradle/invocation/DefaultGradle.java

```diff
@@ -112,7 +116,12 @@ public Path findIdentityPath() {
                     // Not known yet
                     return null;
                 }
-                identityPath = parent.getIdentityPath().child(rootProject.getName());
+                Path parentIdentityPath = parent.findIdentityPath();
```

Comment:
> Make sure `findIdentityPath` does never throw an exception
>
> It was calling parent.getIdentityPath() internally which in turn might
> throw an exception. It is now calling parent.findIdentityPath() instead.

I discovered this through a failing composite build integration test : `IncludedBuildValidationIntegrationTest.reports failure when included build is itself a composite` https://builds.gradle.org/viewLog.html?buildId=2626943

Also it seems it seems with the latest changes to this PR this is no longer a problem here.

I think the new behavior makes more sense, although it might never happen in practice .